srcs/juloo.keyboard2/Keyboard2.java

```diff
@@ -124,6 +125,46 @@ public void onCreate()
     _foldStateTracker.setChangedCallback(() -> { refresh_config(); });
   }
 
+  @Override
+  public boolean onKeyDown(int keyCode, KeyEvent event) {
+    int inputClass = _lastInputType & InputType.TYPE_MASK_CLASS;
+    InputConnection conn = getCurrentInputConnection();
+    if (conn == null || !isInputViewShown() || inputClass != InputType.TYPE_CLASS_TEXT) {
+      return super.onKeyDown(keyCode, event);
+    }
+
+    int orientation = getResources().getConfiguration().orientation;
+    int rotation = ((WindowManager) getSystemService(WINDOW_SERVICE)).getDefaultDisplay().getRotation();
+
+    if (keyCode == KeyEvent.KEYCODE_VOLUME_UP) {
+      if (_keyeventhandler != null) {
+        if (orientation == android.content.res.Configuration.ORIENTATION_LANDSCAPE && rotation == Surface.ROTATION_90) {
+          _keyeventhandler.send_key_down_up(KeyEvent.KEYCODE_DPAD_LEFT); 
+        } else {
+          _keyeventhandler.send_key_down_up(KeyEvent.KEYCODE_DPAD_RIGHT);
+        }
+      }
+      return true;
+    } else if (keyCode == KeyEvent.KEYCODE_VOLUME_DOWN) {
+      if (_keyeventhandler != null) {
+        if (orientation == android.content.res.Configuration.ORIENTATION_LANDSCAPE && rotation == Surface.ROTATION_90) {
+          _keyeventhandler.send_key_down_up(KeyEvent.KEYCODE_DPAD_RIGHT);
+        } else {
+          _keyeventhandler.send_key_down_up(KeyEvent.KEYCODE_DPAD_LEFT);
+        }
+      }
+      return true;
+    }
+    return super.onKeyDown(keyCode, event);
+  }
+
+  @Override
+  public boolean onKeyUp(int keyCode, KeyEvent event) {
+    if (keyCode == KeyEvent.KEYCODE_VOLUME_UP || keyCode == KeyEvent.KEYCODE_VOLUME_DOWN)
```

Comment:
This doesn't do the same `isInputViewShown` and `_lastInputType` checks as `onKeyDown`.